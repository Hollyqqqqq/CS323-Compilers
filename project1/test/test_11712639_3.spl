int a;
int test1(int n){
    if (n < 10){
        a = 0;
        if (n < 5){
            a = 1;
        }
    }else if (n > 10){
        a = 2;
    }
    if (n < 10){
        a = 0;
        if (n < 5){
            a = 1;
        }else{
            a = 3;
        }
    }
}