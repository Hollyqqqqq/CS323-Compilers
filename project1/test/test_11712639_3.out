Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      ExtDecList (1)
        VarDec (1)
          ID: a
      SEMI
    ExtDefList (2)
      ExtDef (2)
        Specifier (2)
          TYPE: int
        FunDec (2)
          ID: test1
          LP
          VarList (2)
            ParamDec (2)
              Specifier (2)
                TYPE: int
              VarDec (2)
                ID: n
          RP
        CompSt (2)
          LC
          StmtList (3)
            Stmt (3)
              IF
              LP
              Exp (3)
                Exp (3)
                  ID: n
                LT
                Exp (3)
                  INT: 10
              RP
              Stmt (3)
                CompSt (3)
                  LC
                  StmtList (4)
                    Stmt (4)
                      Exp (4)
                        Exp (4)
                          ID: a
                        ASSIGN
                        Exp (4)
                          INT: 0
                      SEMI
                    StmtList (5)
                      Stmt (5)
                        IF
                        LP
                        Exp (5)
                          Exp (5)
                            ID: n
                          LT
                          Exp (5)
                            INT: 5
                        RP
                        Stmt (5)
                          CompSt (5)
                            LC
                            StmtList (6)
                              Stmt (6)
                                Exp (6)
                                  Exp (6)
                                    ID: a
                                  ASSIGN
                                  Exp (6)
                                    INT: 1
                                SEMI
                            RC
                  RC
              ELSE
              Stmt (8)
                IF
                LP
                Exp (8)
                  Exp (8)
                    ID: n
                  GT
                  Exp (8)
                    INT: 10
                RP
                Stmt (8)
                  CompSt (8)
                    LC
                    StmtList (9)
                      Stmt (9)
                        Exp (9)
                          Exp (9)
                            ID: a
                          ASSIGN
                          Exp (9)
                            INT: 2
                        SEMI
                    RC
            StmtList (11)
              Stmt (11)
                IF
                LP
                Exp (11)
                  Exp (11)
                    ID: n
                  LT
                  Exp (11)
                    INT: 10
                RP
                Stmt (11)
                  CompSt (11)
                    LC
                    StmtList (12)
                      Stmt (12)
                        Exp (12)
                          Exp (12)
                            ID: a
                          ASSIGN
                          Exp (12)
                            INT: 0
                        SEMI
                      StmtList (13)
                        Stmt (13)
                          IF
                          LP
                          Exp (13)
                            Exp (13)
                              ID: n
                            LT
                            Exp (13)
                              INT: 5
                          RP
                          Stmt (13)
                            CompSt (13)
                              LC
                              StmtList (14)
                                Stmt (14)
                                  Exp (14)
                                    Exp (14)
                                      ID: a
                                    ASSIGN
                                    Exp (14)
                                      INT: 1
                                  SEMI
                              RC
                          ELSE
                          Stmt (15)
                            CompSt (15)
                              LC
                              StmtList (16)
                                Stmt (16)
                                  Exp (16)
                                    Exp (16)
                                      ID: a
                                    ASSIGN
                                    Exp (16)
                                      INT: 3
                                  SEMI
                              RC
                    RC
          RC
